subsys/ipc/ipc_service/backends/ipc_icmsg.c

```diff
@@ -85,6 +85,9 @@ static void mbox_callback_process(struct k_work *item)
 		}
 
 		atomic_set(&dev_data->state, ICMSG_STATE_READY);
+
+		/* Try to read data arriving with magic. */
+		(void)k_work_submit(&dev_data->mbox_work);
```

Comment:
what about joining submit from both cases and having 
```
if (len > 0) {
   (void)k_work_submit(...);
}
```
at the end.

Also would be good to handle error code. Should we assert in case `k_work_submit` returns < 0?

Reply:
done - please have a look again